Lesson 2

hr82al opened the pull request:
Домашняя работа. Исключения.

batiaev commented on Java_Core.Advanced_level.iml:
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
тоже можно добавить в gitignore

hr82al replied:
Спасибо!

hr82al committed:
В main добавлен метод sum, суммирующий все эелементы массива.

hr82al committed:
Update README.md

Описание задачи.

hr82al committed:
Добавлено исключение, проверяющее правильность размера массива.

hr82al committed:
Добавленно исключение проверяющее корректность данных в массиве.

hr82al committed:
Добавлен метод выполняющий суммирование всех элементов массива;
В исключении MyArrayDataException дабавлена строка указывающяя индекс неверного типа данных.

hr82al committed:
В метод sum добавленна поясняющая надпись;
В метод main добавлены премеры неравильных массивов.

hr82al committed:
Исправленно сообщение, выдаваемое исключением MyArrayDataException. А именно изменён формат выводимого сообщения с индексом неверного элемента.С 0-базваого на 1-базовый.

hr82al committed:
Перевод текста в сообщении исключений на английский язык.

batiaev commented on src/main/java/ru/geekbrains/java2/lesson_2/SumArray.java:
```diff
@@ -0,0 +1,81 @@
+        long result = 0;
+        for (String[] strings : _array) {
+            for (String string : strings) {
+                result += (long)Integer.parseInt(string);
```
если хочется использовать long почему не взять Long.parseLong()

batiaev reviewed (commented):
по дз в целом все хорошо, небольшое замечание в коде

hr82al committed:
Изменён метод, извлекающий целое число из строки, с int на long
